packages/sync-actions/src/utils/create-build-actions.js

```diff
@@ -1,7 +1,88 @@
+import isEqual from 'lodash.isequal'
+
 function applyOnBeforeDiff(before, now, fn) {
   return fn && typeof fn === 'function' ? fn(before, now) : [before, now]
 }
 
+function isDateOverlap(
+  oldValidFrom,
+  oldValidUntil,
+  newValidFrom,
+  newValidUntil
+) {
+  if (newValidFrom > oldValidFrom && newValidFrom < oldValidUntil) return true
+  if (newValidUntil < oldValidUntil && newValidUntil > oldValidFrom) return true
+  return false
+}
+
+function getPriceId(newPrice, oldVariantArray) {
+  let newPriceId
+
+  const newPriceComparison = {
```

Comment:
We should likely extract this into a 

```js
const createPriceComparator = (currencyCode, channel, country, customerGroup) => ({
  value: { currencyCode },
  channel,
  country,
  customerGroup,
})
```

and hoist it out of `getPriceId` to not have the VM create new functions on each invocation.